lib/web_ui/test/geometry_test.dart

```diff
@@ -32,33 +34,24 @@ void testMain() {
   test('Offset.fromDirection', () {
     expect(Offset.fromDirection(0.0, 0.0), const Offset(0.0, 0.0));
     expect(Offset.fromDirection(pi / 2.0).dx,
-        closeTo(0.0, 1e-12)); // aah, floating point math. i love you so.
+        moreOrLessEquals(0.0)); // aah, floating point math. i love you so.
     expect(Offset.fromDirection(pi / 2.0).dy, 1.0);
-    expect(Offset.fromDirection(-pi / 2.0).dx, closeTo(0.0, 1e-12));
+    expect(Offset.fromDirection(-pi / 2.0).dx, moreOrLessEquals(0.0));
     expect(Offset.fromDirection(-pi / 2.0).dy, -1.0);
     expect(Offset.fromDirection(0.0), const Offset(1.0, 0.0));
-    expect(Offset.fromDirection(pi / 4.0).dx,
-        closeTo(1.0 / math.sqrt(2.0), 1e-12));
-    expect(Offset.fromDirection(pi / 4.0).dy,
-        closeTo(1.0 / math.sqrt(2.0), 1e-12));
-    expect(Offset.fromDirection(-pi / 4.0).dx,
-        closeTo(1.0 / math.sqrt(2.0), 1e-12));
-    expect(Offset.fromDirection(-pi / 4.0).dy,
-        closeTo(-1.0 / math.sqrt(2.0), 1e-12));
+    expect(Offset.fromDirection(pi / 4.0).dx, moreOrLessEquals(1.0 / math.sqrt(2.0)));
+    expect(Offset.fromDirection(pi / 4.0).dy, moreOrLessEquals(1.0 / math.sqrt(2.0)));
```

Comment:
We also have `within` that supports types other than numbers. For example, it could collapse the above two lines into one. It doesn't have a built-in epsilon value, but we could teach it. We own the implementation.

Reply:
`within` already takes a `distance` parameter which would basically be the epsilon value in this case:
```dart
expect(
  Offset.fromDirection(pi / 4.0),
  within(
    distance: epsilon,
    from: Offset(1.0 / math.sqrt(2.0), 1.0 / math.sqrt(2.0)),
  ),
);
```
Maybe we should make `moreOrLessEquals` generic? Like:
```dart
Matcher moreOrLessEquals<T>(
  T value, {
  double epsilon = precisionErrorTolerance,
  DistanceFunction<T>? distanceFunction,
}) {
  ...
}
```
Then the above two lines become:

```dart
expect(
  Offset.fromDirection(pi / 4.0),
  moreOrLessEquals(Offset(1.0 / math.sqrt(2.0), 1.0 / math.sqrt(2.0))),
);
```

Reply:
If we make `distance` in `within` default to `precisionErrorTolerance`, then it becomes moreOrLessEquals to `moreOrLessEquals`, right?

Reply:
(I would force `AnyDistanceFunction` to return a `double` rather than `num`, but that's just me)

Reply:
Great idea @ditman !